### PR TITLE
DM-37469: Update for ts_tcpip, including using lsst.ts.tcpip.Client

### PR DESCRIPTION
**Pull request by r-owen**

### CODE REVIEW

File: doc/version_history.rst [hunk 1]
```diff
@@ -6,6 +6,20 @@
 Version History
 ###############
 
+v0.31.0
+-------
+
+* `CommandTelemetryClient`: inherit from `lsst.ts.tcpip.Client`, which requires ts_tcpip 1.0.
```
Comment (teweitsai) [inline doc/version_history.rst:12]: inherit --> inherits

File: doc/version_history.rst [hunk 2]
```diff
@@ -6,6 +6,20 @@
...
+* Stop exporting symbols from ts_tcpip, except ``LOCAL_HOST``, which is still used by ts_mtrotator.
```
Comment (teweitsai) [inline doc/version_history.rst:13]: I think `LOCAL_HOST` is used by `ts_m2com` and `ts_mtdome` as well.

**Review by teweitsai — APPROVED** [review]
Looks good!

**Comment by r-owen** [comment]
I revised the code to take advantage of the new wrappers for read and write methods. This was not a huge improvement, but did simplify a few things. I also rewrote a unit test to use async context managers instead of asyncTearDown after running into an issue (which I did fix) of the unit tests hanging. unittest's asyncTearDown method is rather fragile -- if it sees an asyncio.CancelledError the test hangs forever (after finishing teardown). Async context managers don't have this problem.

File: doc/version_history.rst (hunk 2 above)
Comment (r-owen) [reply doc/version_history.rst:14]: Many packages use LOCAL_HOST from ts_tcpip, but I would be surprised if any packages other than ts_mthexrotcomm or ts_mtrotator use it from ts_hexrotcomm.

File: doc/version_history.rst (hunk 1 above)
Comment (r-owen) [reply doc/version_history.rst:12]: I prefer the active voice for release notes.

**Review by teweitsai — APPROVED** [review]
Looks good.

**Pull request merged**